IntroToCSharp/Pages/Lessons/Collections/Arrays.razor

```diff
@@ -0,0 +1,379 @@
+
+
+<Section Title="Introdution">
+    <MudText>
+        An array is a data structure that stores a pre-defined number of variables of the same data type within its bounds.
+         The important part of this definition to take note of is that an array is predefined, meaning that you have to know 
+         how many variables are going to be within the array before you create it.
+         This is because arrays are immutable data structures which means after they are created their <b>length</b> can't be changed.
+    </MudText>
+
+    <Section IsSubSection="true" Title="The Rules">
+        <MudText>When dealing with an array there are four main things to keep in mind:</MudText>
+        <InfoGrid>
+            <GridSquare>Arrays have a <b>pre-defined</b> length. You must give an array a length before you are able to implement it.
+            </GridSquare>
+            <GridSquare>Arrays are <b>immutable</b>. Once the length an array is set, you cannot change it.
+            </GridSquare>
+            <GridSquare>Arrays <b>cannot</b> mix variables types.
+            </GridSquare>
+            <GridSquare>An array's index always begins at 0.
+            </GridSquare>
+        </InfoGrid>
+    </Section>
+</Section>
+
+<Section Title="Our First Array">
+    <MudText>
+        Let's start by making our first array! To do this we are going to start with a string of heroes.
+        To make an array we have to start by declaring what data type that the array will be using followed by
+        a pair of brackets (<Code>[]</Code>) and finally the array's name.
+    </MudText>
+    <MudText>
+        When we set an array's value we "wrap" the value within a pair of curly brackets (<Code>[]</Code>) and seperate
+        each variable with a comma, like so:
+    </MudText>
+
+    <CodeBlock>
+        string[] justiceLeague = {"Superman", "Batman", "Aquaman", "Green Lanter", "Flash", "Wonder Woman", "Martian Manhunter"};
+    </CodeBlock>
+
+    <MudText>
+        As you can see we have a string of 7 superheroes within our array that make up the original Justice League! Each variable
+        within the array is wrapped within the curly brackets and seperate by a comma! Without the comma, the data will either return an error
+        or the data will be read as a single variable!
+    </MudText>
+    <MudText>
+        Another important aspect to understand an Array is how their index works. An array's index is the position a variable is after
+        within an array. Rather than starting at 1 and counting up, arrays always start and 0 and count up. For our justiceLeague array
+        this means that "Superman" is at the 0 index of the array or in coding terms:
+    </MudText>
+
+    <CodeBlock>
+        string[] justiceLeague = {"Superman", "Batman", "Aquaman", "Green Lanter", "Flash", "Wonder Woman", "Martian Manhunter"};
+        Console.WriteLine("The first superhero in our array is " + justiceLeague[0]);
+    </CodeBlock>
```

Comment:
I would move these two paragraphs into a new section called `Reading Data from an Array`.

Reply:
In the Reading Data from an array section, talk about index out of bounds.